src/getChildEventSubscriber.js

```diff
@@ -55,110 +59,58 @@ export default function getChildEventSubscriber(addListener, key) {
       const { state, lastState, action } = payload;
       const lastRoutes = lastState && lastState.routes;
       const routes = state && state.routes;
+
       const lastFocusKey =
         lastState && lastState.routes && lastState.routes[lastState.index].key;
       const focusKey = routes && routes[state.index].key;
 
       const isFocused = focusKey === key;
-      const wasFocused = lastFocusKey === key;
       const lastRoute =
         lastRoutes && lastRoutes.find(route => route.key === key);
       const newRoute = routes && routes.find(route => route.key === key);
-      const eventContext = payload.context || 'Root';
       const childPayload = {
-        context: `${key}:${action.type}_${eventContext}`,
+        context: `${key}:${action.type}_${payload.context || 'Root'}`,
         state: newRoute,
         lastState: lastRoute,
         action,
         type: eventName,
       };
-
       const isTransitioning = !!state && state.isTransitioning;
-      const wasTransitioning = !!lastState && lastState.isTransitioning;
-      const didStartTransitioning = !wasTransitioning && isTransitioning;
-      const didFinishTransitioning = wasTransitioning && !isTransitioning;
-      const wasChildFocused = isChildFocused;
-      if (eventName !== 'action') {
-        switch (eventName) {
-          case 'didFocus':
-            isParentFocused = true;
-            break;
-          case 'didBlur':
-            isParentFocused = false;
-            break;
-        }
-        if (isFocused && eventName === 'willFocus') {
-          emit(childPayload);
-        }
-        if (isFocused && !isTransitioning && eventName === 'didFocus') {
-          emit(childPayload);
-          isChildFocused = true;
-        }
-        if (isFocused && eventName === 'willBlur') {
-          emit(childPayload);
+
+      const previouslyLastEmittedEvent = lastEmittedEvent;
+
+      if (lastEmittedEvent === 'didBlur') {
+        if (eventName === 'willFocus' && isFocused) {
+          emit((lastEmittedEvent = 'willFocus'), childPayload);
+        } else if (eventName === 'action' && isFocused) {
+          emit((lastEmittedEvent = 'willFocus'), childPayload);
         }
-        if (isFocused && !isTransitioning && eventName === 'didBlur') {
-          emit(childPayload);
+      }
+      if (lastEmittedEvent === 'willFocus') {
+        if (eventName === 'didFocus' && isFocused && !isTransitioning) {
+          emit((lastEmittedEvent = 'didFocus'), childPayload);
+        } else if (!isTransitioning && isFocused && eventName === 'action') {
```

Comment:
let's keep the same order for the conditionals here, 

`if (eventName === 'didFocus' && isFocused && !isTransitioning)` ✅ 
`else if (!isTransitioning && isFocused && eventName === 'action')` 🚫
`else if (eventName === 'action' && isFocused && !isTransitioning)` ✅

Reply:
(same applies for other conditionals around here)